src/system_reader.rs

```diff
@@ -1100,6 +1113,43 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn shared_symbol_table_import_binary() -> IonResult<()> {
+        let mut map_catalog = MapCatalog::new();
+        // add a table with system symbol `name` and a new symbol `foo`
+        // reader should still add this system symbol `name` again($10), but whenever writing this symbol
+        // smallest ID($4) will be used
+        map_catalog.insert_table(SharedSymbolTable::new("shared_table", 1, ["name", "foo"])?);
+        // The stream contains a shared symbol table import
+        let mut reader = binary_system_reader_with_catalog_for(
+            &[
+                0xe0, 0x01, 0x00, 0xea, // Ion 1.0 Version Marker
+                0xee, 0x9d, 0x81, 0x83, // '$ion_symbol_table'::
+                0xde, 0x99, // 26 bytes struct
+                0x86, // Field $6 `imports`
+                0xbe, 0x96, // List
+                0xde, 0x94, // 21 bytes struct
+                0x84, // Field $4 `name`
+                0x8c, 0x73, 0x68, 0x61, 0x72, 0x65, 0x64, 0x5f, 0x74, 0x61, 0x62, 0x6c,
+                0x65, // "shared_table"
+                0x85, // Field $5 `version`
+                0x21, 0x01, // INT 1
+                0x88, // $8 `max_id`
+                0x21, 0x02, // INT 2
+                0x71, 0x04, // $4 `name`
+                0x71, 0x0b, // $11 `foo`
```

Comment:
Please also have this test that `$10` turns into `name`. The comment mentions that it should be added, but I don't think it's being tested here.